### PR TITLE
Try adding nextflow strict

### PR DESCRIPTION
**Pull request by jfy133**
Experimentation following: https://nfcore.slack.com/archives/C043UU89KKQ/p1709835715768389

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/taxprofiler/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/taxprofiler _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by jfy133** [comment]
OK I'm getting some weird variation which I'm still investigating, but for now...

**Comment by jfy133** [comment]
OK it magically has disappeared... we will have to monitor this... basically some of the `db_params` were being emitted as `null` and so `split()` would stop working... I replicated it a couple of tiemes ion my laptop but now it doesn't happen.

**Comment by LilyAnderssonLee** [comment]
I am also confused by this 'null' error. Do you know if this issue occurs for anyone else?

**Review by LilyAnderssonLee — APPROVED** [review]
Well done! I tested it locally and it performed exactly as expected. LGTM!

File: workflows/taxprofiler.nf [hunk 1]
```diff
@@ -179,14 +179,12 @@ workflow TAXPROFILER {
...
-        .map { db_meta, db -> [ db_meta.db_params ]
-            def corrected_db_params = db_meta.db_params == null ? '' : db_meta.db_params
-            db_meta.db_params = corrected_db_params
-            [ db_meta, db ]
-        }
-
+    ch_final_dbs = ch_dbs_for_untar.skip
+                    .mix( ch_outputdb_from_untar  )
+                    .map { db_meta, db ->
+                        def corrected_db_params = db_meta.db_params == null ? [ db_params: '' ] : [ db_params: db_meta.db_params ]
```
Comment (Midnighter) [inline workflows/taxprofiler.nf:185]: You can use a "truthy" value check here. It's not really better or worse, though.

```suggestion
                        def corrected_db_params = db_meta.db_params ? [ db_params: db_meta.db_params ] : [ db_params: '' ]
```

**Comment by LilyAnderssonLee** [comment]
Not sure why there is a `krakentools` folder in the results, and uncertain about when this error was introduced. 
Here's the code to reproduce the result: `nextflow run main.nf -profile singularity,test --outdir results`.

**Comment by jfy133** [comment]
@LilyAnderssonLee you're right it came with the template, I guess there is now a default folder name for all modules. I'll have a look to see if they are relevant :)

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### `Added`
 
 - [#454](https://github.com/nf-core/taxprofiler/pull/454) Updated to nf-core pipeline template v2.13.1 (added by @LilyAnderssonLee & @sofstam)
+- [#461](https://github.com/nf-core/taxprofiler/pull/461) Turned on 'strict' Nextflow evaluation runs (added by @jfy133)
+- [#461](https://github.com/nf-core/taxprofiler/pull/461) Optimised database compression so each compressed input database is untarred once, and shared amongst each run with different parameters (added by @jfy133)
```
Comment (jfy133) [inline CHANGELOG.md:12]:
```suggestion
- [#461](https://github.com/nf-core/taxprofiler/pull/461) Optimised database compression so each compressed input database is untarred once, and shared amongst each run with different parameters (added by @jfy133)
[#461](https://github.com/nf-core/taxprofiler/pull/461) Added new parameter to optionally save uncompressed databases (added by @jfy133)
```

**Comment by jfy133** [comment]
@nf-core-bot fix linting

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+  [#461](https://github.com/nf-core/taxprofiler/pull/461) Added new parameter to optionally save uncompressed databases (added by @jfy133)
```
Comment (LilyAnderssonLee) [inline CHANGELOG.md:13]:
```suggestion
- [#461](https://github.com/nf-core/taxprofiler/pull/461) Added new parameter to optionally save uncompressed databases (added by @jfy133)
```

File: nextflow_schema.json [hunk 4]
```diff
@@ -34,6 +34,12 @@
...
                     "help_text": "You will need to create a design file with information about the samples in your experiment before running the pipeline. Use this parameter to specify its location. It has to be a comma-separated file with 4 columns, and a header row. See [usage docs](https://nf-co.re/taxprofiler/dev/usage#full-database-sheet).\n\nProfilers will only be executed if a corresponding  database are supplied. \n\nWe recommend storing this database sheet somewhere centrally and accessible by others members of your lab/institutions, as this file will likely be regularly reused."
                 },
+                "save_untarred_databases": {
+                    "type": "boolean",
+                    "fa_icon": "fas fa-database",
+                    "description": "Specify to save decompressed user-supplied TAR archives of databases",
+                    "help_text": "If input databases are supplied as gzipped TAR archives, in some cases you may want to move and re-use these for future runs./n/n Specifying this parameter will save these to `--outdir results/` under a directory called `untar`."
```
Comment (LilyAnderssonLee) [inline nextflow_schema.json:41]:
```suggestion
                    "help_text": "If input databases are supplied as gzipped TAR archives, in some cases you may want to move and re-use these for future runs. Specifying this parameter will save these to `--outdir results/` under a directory called `untar`."
```

**Comment by LilyAnderssonLee** [comment]
Tested locally and worked well. Good job!

**Pull request merged**